### DISCUSSION TITLE
Blazorise Guidance

### DISCUSSION THREAD

**Discussion by markdav-is** [oqtane/oqtane.framework#701] [General] [answered]
We are exploring using Blazorize as a control and style library.  https://blazorise.com/docs/start/ I'm curious if others have gone down this road and have any tips to share. My initial feeling is that installing this goes beyond theming and requires lots of modifications to the core framework.  I'm concerned that I'll be making upgrades harder on myself later.

**Comment by mikecasas** [comment]
Checkout issues #690 and #693.

But it does seem that Blazorise needs more assets added.

**Reply by markdav-is** [reply]
thanks.  I'm watching a thread about modules and assets and think we'll be able to get this rolling.

**Comment by markdav-is** [comment]
I've created a Blazorize service in my module client using IService bit I don't think it's getting initialized. 

![image](https://user-images.githubusercontent.com/311063/93912296-6d877f80-fcb8-11ea-8221-74232c18bdc5.png)

**Reply by hishamco** [reply]
What are you intend to do, the proper place to add services is in `Startup` or `Programs`

**Reply by markdav-is** [reply]
modules don't have startup or program so I'm registering the service using the IService interface.  Oqtane enumerates all types in the modules and registers those that implement IService.  you can see this happening in AddOqtaneServices()  in OqtaneServiceCollectionExtensions.

**Reply by hishamco** [reply]
Regarding to the `AddOqtaneServices()` it extend `IServiceCollection` but not injected

Please let me know what you wanna do, because the issue is not clear to me

**Reply by markdav-is** [reply]
I'm trying to add blazorize to my module.  To do that I need to add the blazorize services.  This is true for any module that has service dependencies.  The way you add services to Oqtane is to implement the IService interface.  You can see the getting started docs here: https://blazorise.com/docs/start/   so I need to "hook in" to the service registration and add the blazorize services. So, if IServiceCollection is not injected, can I get to it from something that is injected?

**Reply by hishamco** [reply]
Could please have a look to [Oqtane Blog module](https://github.com/oqtane/oqtane.blogs), you should register your services in Oqtane.Server

**Reply by sbwalker** [reply]
@nohorse The IService interface for auto registration is really only intended for simple API/Repository services that you will create for your module. For more elaborate third party libraries such as Blazorize, in order to register these services you either need to modify the core framework Startup class ( not recommended for obvious reasons ) or use the IClientStartup/IServerStartup interfaces which allow you to inject custom registration code into the framework. The Blog module does not contain an example of this. I know that @mikecasas is successfully using IServerStartup so perhaps he can provide an example.

**Reply by markdav-is** [reply]
Thanks @sbwalker I'll look into IClientStartup as this is a client UI framework.

**Reply by sbwalker** [reply]
@nohorse you will need to understand the runtime characteristics of Blazor in order to understand the startup interfaces. If you are running on Blazor Server then you need to register services in IServerStartup and if you are running on Blazor WebAssembly you need to register services in IClientStartup ( and if you want to run on either then you need to implement both ).

**Reply by hishamco** [reply]
I think similar issue discussed here #693

**Comment by markdav-is** [comment, marked as answer]
the answer on this one was covered in https://github.com/oqtane/oqtane.framework/discussions/690
I ended up using both IServerStartup and IHostResources 
```
BlazorizeService : IServerStartup, IHostResources
```
